### DISCUSSION TITLE
Is AzuraCast PaaS-friendly?

### DISCUSSION THREAD

**Discussion by tran-hai-long** [AzuraCast/AzuraCast#6183] [Q&A] [answered]
Hello everyone, i'm thinking of deploying AzuraCast on a PaaS like Heroku, Fly.io or Railway.app
I admit, I'm a complete noob when it comes to managing a VPS. That's why I want to deploy it on a PaaS and let the professionals handle the OS maintenance. 
Most PaaS do not allow you to ssh or run bash, instead will automatic build a Docker image based on your Dockerfile. I want to know if it's sufficient to get AzuraCast up and running, or do you absolutely have to run docker.sh for proper setup?

**Comment by BusterNeece** [comment]
It is indeed possible to host AzuraCast on some of these services (I've tried it with Fly.io), but while you can do it, the question is whether you should do so. These services are meant to run certain on-demand workloads (things like web applications), and if you run a constant-compute workload like audio transcoding on them, it's very likely that the host will detect this and throttle you rather quickly.

It's definitely recommended to run AzuraCast on a service that provides dedicated-CPU VPSes, so this isn't an issue.

**Reply by tran-hai-long** [reply]
> It is indeed possible to host AzuraCast on some of these services (I've tried it with Fly.io), but while you can do it, the question is whether you should do so. These services are meant to run certain on-demand workloads (things like web applications), and if you run a constant-compute workload like audio transcoding on them, it's very likely that the host will detect this and throttle you rather quickly.
> 
> It's definitely recommended to run AzuraCast on a service that provides dedicated-CPU VPSes, so this isn't an issue.

Thank you for your input. I looked at Fly.io pricing page and seems like they do offer Dedicated CPU for 30$/month.
Assuming CPU power is not a problem, can you write an idiot-proof guide on how to deployt AzuraCast on one of those PaaS? I tried to look inside docker.sh but I am not familiar with bash so it looks kinda cryptic to me.

**Comment by BusterNeece** [comment, marked as answer]
I looked into this, and while it's certainly possible, it's probably not quite as simple as we thought, especially since if you set up a database, these services kind of want you to use _their_ database, not the one we bundle with our app.

We can revisit this issue in the future, though; I'd recommend submitting a feature request at https://features.azuracast.com/ just so we can keep track of public interest on it.
